### PR TITLE
use transaction per datasource

### PR DESCRIPTION
**Pull request by maggarwal13**

### CODE REVIEW

File: whois-commons/src/main/resources/applicationContext-commons.xml [hunk 1]
```diff
@@ -13,69 +13,6 @@
                            http://www.springframework.org/schema/cache
                            http://www.springframework.org/schema/cache/spring-cache.xsd">
 
-    <!-- TODO: this is loaded many many times on context init, which slows down startup process -->
-    <!-- @Transactional CAN be used on private methods with mode="aspectj". However, annotations on interfaces will NOT work! -->
-    <tx:annotation-driven mode="aspectj"/>
```
Comment (eshryane) [inline whois-commons/src/main/resources/applicationContext-commons.xml:18]: Should be replaced by `EnableTransactionManagement` already, correct?

File: whois-commons/src/main/resources/applicationContext-commons.xml [hunk 2]
```diff
@@ -13,69 +13,6 @@
...
-
-    <context:spring-configured/>
```
Comment (eshryane) [inline whois-commons/src/main/resources/applicationContext-commons.xml:20]: Should be replaced by EnableSpringConfigured annotation, correct?

File: whois-commons/src/main/resources/applicationContext-commons.xml [hunk 3]
```diff
@@ -13,69 +13,6 @@
...
     <context:component-scan base-package="net.ripe.db.whois.common"/>
 
-    <context:annotation-config/>
```
Comment (eshryane) [inline whois-commons/src/main/resources/applicationContext-commons.xml:23]: Do we need `AnnotationConfigApplicationContext` to replace `<context:annotation-config>` ?
i.e. to support non-Spring (Jakarta) annotations

File: whois-commons/src/main/resources/applicationContext-commons.xml [hunk 4]
```diff
@@ -13,69 +13,6 @@
...
-    <cache:annotation-driven/>
```
Comment (eshryane) [inline whois-commons/src/main/resources/applicationContext-commons.xml:24]: Should be replaced by Spring `EnableCaching` annotation ?

File: whois-commons/src/main/resources/applicationContext-commons.xml (hunk 1 above)
Comment (maggarwal13) [reply whois-commons/src/main/resources/applicationContext-commons.xml:18]: Yes, already there in WhoisServer
@EnableSpringConfigured
@EnableTransactionManagement(mode = AdviceMode.ASPECTJ)
@EnableCaching(mode = AdviceMode.ASPECTJ)

File: whois-commons/src/main/resources/applicationContext-commons.xml (hunk 2 above)
Comment (maggarwal13) [reply whois-commons/src/main/resources/applicationContext-commons.xml:20]: Yes, already there in WhoisServer
@EnableSpringConfigured
@EnableTransactionManagement(mode = AdviceMode.ASPECTJ)
@EnableCaching(mode = AdviceMode.ASPECTJ)

File: whois-commons/src/main/resources/applicationContext-commons.xml (hunk 4 above)
Comment (maggarwal13) [reply whois-commons/src/main/resources/applicationContext-commons.xml:24]: Yes, already there in WhoisServer

File: whois-commons/src/main/resources/applicationContext-commons.xml (hunk 3 above)
Comment (maggarwal13) [reply whois-commons/src/main/resources/applicationContext-commons.xml:23]: applicationContext-whois-base.xml has this already <context:annotation-config>

**Review by eshryane — APPROVED** [review]

**Comment by eshryane** [comment]
Transaction Manager is not specified on every Transactional annotation, because the default is still the transaction manager for the sourceAwareDataSource (marked with a Primary annotation), and that can still be used in most places.

**Comment by maggarwal13** [comment]
IT tests passed

**Pull request merged**